### PR TITLE
Enable Jenkins Security Scan

### PR DESCRIPTION
**Pull request by strangelookingnerd**
It is stated in the `JENKINS SECURITY POLICY` that

> The Jenkins project takes security seriously. We make every possible effort to ensure users can adequately secure their automation infrastructure. To that end, we work with Jenkins core and plugin developers, as well as security researchers, to fix security vulnerabilities in Jenkins in a timely manner, and to improve the security of Jenkins in general.

We as plugin developers should try our best to support this. One way of doing so is to enable the Jenkins Security Scan on this repository. For more details see the [documentation](https://www.jenkins.io/doc/developer/security/scan/).

This PR may create new issues in the Code scanning section of this project once it is merged. I'd be happy to try and help resolve those if necessary.

_Please note this PR was created semi-automatically. If you find any issue with it, don't hesitate to ping me._

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: .github/workflows/jenkins-security-scan.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  security-events: write
+  contents: read
+  actions: read
+
+jobs:
+  security-scan:
+    uses: jenkins-infra/jenkins-security-scan/.github/workflows/jenkins-security-scan.yaml@v2
+    with:
+      java-cache: 'maven' # Optionally enable use of a build dependency cache. Specify 'maven' or 'gradle' as appropriate.
+      java-version: 17 # What version of Java to set up for the build.
```
Comment (rantoniuk) [inline .github/workflows/jenkins-security-scan.yml:22]: remove the default - [17 is used by default](https://github.com/jenkins-infra/jenkins-security-scan/blob/main/.github/workflows/jenkins-security-scan.yaml#L33)

**Comment by rantoniuk** [comment]
The build is failing ATM, can you take a look?

File: .github/workflows/jenkins-security-scan.yml (hunk 1 above)
Comment (strangelookingnerd) [reply .github/workflows/jenkins-security-scan.yml]: Yeah, this was changed after I created the PR. I have updated the config.

**Comment by strangelookingnerd** [comment]
> The build is failing ATM, can you take a look?

There was a Connection Reset during Maven dependency resolution. I have seen this happen in other places, too. Once the actions cache is being created it should no longer appear.   Otherwise it is usually fixed by re-running the check.

**Pull request merged**